src/schoology-extractor/schoology_extractor/api/request_client.py

```diff
@@ -167,13 +182,22 @@ def post(self, resource: str, json: dict) -> dict:
         RuntimeError
             If the POST operation is unsuccessful.
         """
+        url = f"{self.base_url}{resource}"
         response = self.oauth.post(
-            url=f"{self.base_url}{resource}",
+            url=url,
             headers=self._request_header,
             auth=self.oauth.auth,
             json=json,
         )
 
+        if response.status_code == HTTPStatus.TOO_MANY_REQUESTS:
+            logger.warn(
+                "Hit API rate limiting on POST to %s, will retry with backoff", url
+            )
+            raise HTTPError(
+                f"{response.reason} ({response.status_code}): {response.text}"
+            )
+
```

Comment:
Would it make sense to extract a function for this block of code, to avoid duplication?